File: modulo1/programas/ex2SomaAcumulada.py
def soma_progressiva(n):
    s = 0.0
    for i in range(1, n+1):
        s += 1.0 / (i * i)
    return s

def soma_regressiva(n):
    s = 0.0
    for i in reversed(range(1, n+1)):
        s += 1.0 / (i * i)
    return s

n = 1000000
exato = 1.6449340668482264

prog = soma_progressiva(n)
regr = soma_regressiva(n)

print(f"Soma progressiva: {prog:.15f}")
print(f"Soma regressiva: {regr:.15f}")
print(f"Erro progressivo: {abs(prog - exato):.3e}")
print(f"Erro regressivo: {abs(regr - exato):.3e}")

File: modulo1/programas/ex2SomaAcumulada.cpp
#include <iostream>
#include <iomanip>
#include <cmath>
using namespace std;

int main() {
    int n = 1000000;
    double soma_prog = 0.0, soma_regr = 0.0;
    double exato = 1.6449340668482264;
    for (int i = 1; i <= n; ++i) {
        double ii = static_cast<double>(i);
        soma_prog += 1.0 / (ii * ii);
    }

    for (int i = n; i >= 1; --i) {
        double ii = static_cast<double>(i);
        soma_regr += 1.0 / (ii * ii);
    }

    cout << fixed << setprecision(15);
    cout << "Soma progressiva: " << soma_prog << endl;
    cout << "Soma regressiva: " << soma_regr << endl;
    cout << "Erro progressivo: " << fabs(soma_prog - exato) << endl;
    cout << "Erro regressivo: " << fabs(soma_regr - exato) << endl;

    return 0;
}
